### PR TITLE
[docs] Flesh out index page

### PR DESCRIPTION
**Pull request by benraskin92**
Screen Shot 2019-04-16 at 9 55 08 AM

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -2,6 +2,35 @@
...
 
-[m3]: https://github.com/m3db/m3
+## Key Features
+
+Below are some of the key features of M3 that make it a great platform for time series data.
+
+* Distributed time series database (M3DB) that provides scalable storage and a reverse index of time series.
+* Side-car process (M3Coordinator) that allows M3DB to act as the long-term storage for Prometheus.
+* Distributed query engine (M3Query) with native support for PromQL and Graphite.
+* Aggregation tier (M3Aggregator) that runs as a dedicated metrics aggregator.
```
Comment (richardartoul) [inline docs/index.md]: "runs as a dedicated metrics aggregator / downsampler. This enables the ability to aggregate metrics stored at various retentions at different resolutions."

File: docs/index.md [hunk 2]
```diff
@@ -2,6 +2,35 @@
...
+
+## Getting Started
```
Comment (richardartoul) [inline docs/index.md]: Add: "Make sure to read our Operational Guides (link) before running in production!"

File: docs/index.md [hunk 3]
```diff
@@ -2,6 +2,35 @@
 
 ## About
 
-M3 is a metrics platform that is built on M3DB, a distributed timeseries database. The project monorepo can be found [here][m3].
+[M3](https://github.com/m3db/m3) is an open source metrics platform that was developed at Uber out of necessity.
+After using what was available as open source and finding we were unable to use them at our scale due to issues
```
Comment (mway) [inline docs/index.md]: avoid first-person where possible. i'd recommend something like:

> After using open-source metrics solutions and finding issues with them at scale – such as reliability, cost, and operational complexity – M3 was created from the ground up to provide Uber with a native, distributed time series database, a highly-dynamic and performant aggregation service, a query engine, and other supporting infrastructure.

File: docs/index.md [hunk 4]
```diff
@@ -2,6 +2,35 @@
...
+with their reliability, cost and operationally intensive nature, we built our own metrics platform piece by piece.
+We used our experience to help us build a native distributed time series database, a highly dynamic and performant
+aggregation service, query engine and other supporting infrastructure.
 
-[m3]: https://github.com/m3db/m3
+## Key Features
+
+Below are some of the key features of M3 that make it a great platform for time series data.
```
Comment (mway) [inline docs/index.md]: > M3 has several features, provided as discrete components, which make it an ideal platform for time series data at scale:

File: docs/index.md [hunk 5]
```diff
@@ -2,6 +2,35 @@
...
+
+* Distributed time series database (M3DB) that provides scalable storage and a reverse index of time series.
```
Comment (mway) [inline docs/index.md]: > A distributed time series database, M3DB, that provides scalable storage for time series data and a reverse index.

also, link "distributed time series database, M3DB" to the m3db overview docs

File: docs/index.md [hunk 6]
```diff
@@ -2,6 +2,35 @@
...
+* Side-car process (M3Coordinator) that allows M3DB to act as the long-term storage for Prometheus.
```
Comment (mway) [inline docs/index.md]: > A sidecar process, M3Coordinator, [...]

also link this in the same way as m3db

File: docs/index.md [hunk 7]
```diff
@@ -2,6 +2,35 @@
...
+* Distributed query engine (M3Query) with native support for PromQL and Graphite.
```
Comment (mway) [inline docs/index.md]: > A distributed query engine, M3Query, [...]

same as above

Comment (mway) [reply docs/index.md]: also, there's no world in which we don't eventually add M3QL - let's find a good way to call that out, even if it's something like

> [...] PromQL and Graphite (M3QL support forthcoming).

File: docs/index.md (hunk 1 above)
Comment (mway) [inline docs/index.md]: > An aggregation tier, M3Aggregator, [...]

linked as above, and with richie's feedback

File: docs/index.md (hunk 2 above)
Comment (mway) [reply docs/index.md]: +1

File: docs/index.md [hunk 8]
```diff
@@ -2,6 +2,35 @@
...
+* Aggregation tier (M3Aggregator) that runs as a dedicated metrics aggregator.
+
+## Getting Started
+
+Getting started with M3 is as easy as following one of our `How-To` guides.
```
Comment (mway) [inline docs/index.md]: s/our/the/

also, nit: "How-To" is fine without preformatting

File: docs/index.md [hunk 9]
```diff
@@ -2,6 +2,35 @@
...
+
+* [Single M3DB node deployment](how_to/single_node.md)
+* [Clustered M3DB deployment](how_to/cluster_hard_way.md)
+* [M3DB on Kubernetes](how_to/kubernetes.md)
+* [Isolated M3Query on deployment](how_to/query.md)
+
+## Support
+
+Should you run into any issues, have questions about M3, or want to leave any comments, you can reach us
```
Comment (mway) [inline docs/index.md]: > For support with any issues, questions about M3 or its operation, or to leave any comments, the team can be reached in a variety of ways:

**Pull request merged**